flake.nix

```diff
@@ -111,6 +147,57 @@
                 command "nix-shell"
               }.
             '';
+
+          o1js-npm-deps = pkgs.buildNpmPackage
+            { name = "o1js";
+              src = with pkgs.lib.fileset;
+                  (toSource {
+                    root = ./.;
+                    fileset = unions [
+                      ./package.json
+                      ./package-lock.json
+                    ];
+                  });
+              # If you see 'ERROR: npmDepsHash is out of date' in ci
+              # set this to blank run ``nix build o1js#o1js-bindings`
+              # If you don't want to install nix you can also set it to "" and run ci to get the new hash
```

Comment:
Unfortunately Sometimes yes.
 
It depends on rather you have built it with the previous hash on your system. It uses the npmDepsHash to cache the downloads so if you've built it with that hash before it just uses that, but then it checks those downloads match the package.json in case the npmDepsHash is out of date. Ideally when they don't match it would download them get the hash and tell you what it is but unfortunately it just tells you that it's wrong so to get the correct hash you have to set it to a hash that won't have been built (ie. "" which internally gets expanded to `sha256-AAAAA...`) then it tries to build it gets a different hash and tells you the new hash.